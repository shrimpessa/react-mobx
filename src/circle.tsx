export const Circle = (props: { width: number, height: number, color: string, children: any}) => {
    return (
        <div style={{
            backgroundColor: props.color,
            borderRadius: 200,
            margin: 30,
            width: props.width,
            height: props.height
        }}>
            <div style={{margin: 140}}>
                {props.children}
            </div>
        </div>
    );
}