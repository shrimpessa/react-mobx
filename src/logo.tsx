export const Logo = (props: { color: string }) => (
    <svg width={70} height={70} viewBox="0 0 13 12" fill="none">
        <path
            d="M12.48 3.624l.002.017.003.017c.005.033.067.449-.104 1.164h0c-.244 1.03-.81 1.973-1.646 2.725L6.464 11.33 2.267 7.548h0C1.43 6.796.863 5.852.618 4.822h0c-.17-.716-.11-1.13-.104-1.163l.003-.018.002-.017C.69 1.757 1.996.5 3.54.5c1.026 0 1.949.543 2.493 1.468l.426.724.434-.72A3.002 3.002 0 019.46.5c1.544 0 2.85 1.258 3.02 3.124z"
            stroke='#fff'
            fill={props.color}
        />
    </svg>
);