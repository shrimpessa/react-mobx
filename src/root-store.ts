import { makeAutoObservable } from "mobx";

class Store {
    circleBackgroundColor: string = '#F2E4EB';
    logoColors: {
        first: string,
        second: string,
        third: string,
        fourth: string,
        fifth: string,
        sixth: string
    } = {
        first: '#D5A6BD',
        second: '#A6D5BE',
        third: '#D5A6BD',
        fourth: '#A6D5BE',
        fifth: '#D5A6BD',
        sixth: '#A6D5BE'
    }

    constructor() {
        makeAutoObservable(this);
    }

}

export const rootStore = new Store();