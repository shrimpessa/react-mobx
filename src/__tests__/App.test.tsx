import React from 'react';
import { Circle } from "../circle";
import { Logo } from "../logo";
import { rootStore } from "../root-store";

describe('Render icons', () => {
  test('Render Circle', () => {
    const output = <Circle color={'#fff'} width={33} height={33} children={{}}/>;
    expect(output).toMatchSnapshot();
  });

  test('Render Logo', () => {
    const output = <Logo color={'#fff'} />;
    expect(output).toMatchSnapshot();
  });
});

describe('Test mixColors', () => {
  test('mixColors: 4 > result > 0', () => {
    const result = rootStore.getRandomInt(3);
    expect(result > 0 && result < 4).toBeTruthy()
  });

  test('mixColors result is int', () => {
    const result = rootStore.getRandomInt(3);
    expect(Number.isInteger(result)).toBeTruthy();
  });
});