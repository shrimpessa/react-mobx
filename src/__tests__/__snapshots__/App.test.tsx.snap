// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render icons Render Circle 1`] = `
<Circle
  color="#fff"
  height={33}
  width={33}
>
  Object {}
</Circle>
`;

exports[`Render icons Render Logo 1`] = `
<Logo
  color="#fff"
/>
`;
